packages/aws-cdk-lib/aws-appsync/lib/eventapi.ts

```diff
@@ -0,0 +1,584 @@
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * an unique AWS AppSync Event API identifier
+   * i.e. 'lxz775lwdrgcndgz3nurvac7oa'
+   */
+  public readonly apiId: string;
+
+  /**
+   * the ARN of the API
+   */
+  public readonly apiArn: string;
+
+  /**
+   * the domain name of the API
+   */
+  public readonly dns: IResolvable;
+
+  /**
+   * The Authorization Types for this Event Api
+   */
+  public readonly authProviderTypes: AuthorizationType[];
+
+  /**
+   * The default publish auth modes for this Event Api
+   */
+  public readonly defaultPublishModeTypes: AuthorizationType[];
+
+  /**
+   * The default subscribe auth modes for this Event Api
+   */
+  public readonly defaultSubscribeModeTypes: AuthorizationType[];
+
+  /**
+   * the configured API keys, if present
+   *
+   * @default - no api key
+   * @attribute ApiKeys
+   */
+  public readonly apiKeys: { [key: string]: CfnApiKey } = {};
+
+  /**
+   * the CloudWatch Log Group for this API
+   */
+  public readonly logGroup: ILogGroup;
+
+  private api: CfnApi;
+  private eventConfig: CfnApi.EventConfigProperty;
+  private domainNameResource?: CfnDomainName;
+
+  constructor(scope: Construct, id: string, props: EventApiProps) {
+    if (props.apiName !== undefined && !Token.isUnresolved(props.apiName)) {
+      if (props.apiName.length < 1 || props.apiName.length > 50) {
+        throw new Error(`\`apiName\` must be between 1 and 50 characters, got: ${props.apiName.length} characters.`);
+      }
+
+      const apiNamePattern = /^[A-Za-z0-9_\-\ ]+$/;
+
+      if (!apiNamePattern.test(props.apiName)) {
+        throw new Error(`\`apiName\` must contain only alphanumeric characters, underscores, hyphens, and spaces, got: ${props.apiName}`);
+      }
+    }
+
+    super(scope, id, {
+      physicalName: props.apiName ?? Lazy.string({
+        produce: () =>
+          Names.uniqueResourceName(this, {
+            maxLength: 50,
+            separator: '-',
+          }),
+      }),
+    });
+
+    this.validateOwnerContact(props.ownerContact);
+
+    this.authProviderTypes = this.setupAuthProviderTypes(props.authorizationConfig?.authProviders);
+
+    const authProviders = props.authorizationConfig?.authProviders ?? [{ authorizationType: AuthorizationType.API_KEY }];
+    this.validateAuthorizationProps(authProviders);
+
+    const connectionAuthModeType = props.authorizationConfig?.connectionAuthModeTypes ?? this.authProviderTypes;
+    const defaultPublishAuthModeTypes = props.authorizationConfig?.defaultPublishAuthModeTypes ?? this.authProviderTypes;
+    const defaultSubscribeAuthModeTypes = props.authorizationConfig?.defaultSubscribeAuthModeTypes ?? this.authProviderTypes;
+
+    this.validateAuthorizationConfig(authProviders, connectionAuthModeType);
+    this.validateAuthorizationConfig(authProviders, defaultPublishAuthModeTypes);
+    this.validateAuthorizationConfig(authProviders, defaultSubscribeAuthModeTypes);
```

Comment:
nit: I see that there are a few validations we perform, i.e. owner contract, auth modes, etc. IMO, we should have a overall `validateConfiguration` method (or similar name) and in the method, we can call different validate methods.